content/blog/how-to-ask-good-questions/index.md

```diff
@@ -0,0 +1,113 @@
+	attached: [],
+	license: 'cc-by-nc-sa-4'
+}
+---
+
+# How to ask good questions
+
+## Introduction
+
+In this day and age, the programming ecosystem has become so rich and complex that asking questions is inevitable for any developer, regardless of what stage of their career they find themselves in.
+
+However, it is not always easy to ask a good question, and often you'll see StackOverflow questions being downvoted or taken down due to the question being of poor quality.
+
+Asking good questions is essential to getting help quickly, as whoever is reading your question will have immediate access to most of the information they need to come up with an answer that could potentially answer your doubts.
+
+So in this article, I wanted to cover the basics on how to ask a good question, in order to get good answers.
+
+Note: Most of these concepts could also apply to opening issues on OSS projects, or to give feedback to different pieces of software
+
+## Define the exact problem you're facing
+
+Now whilst this sounds simple, there are many people who struggle to define their problem. If you're not able to identify the problem you've got, how do you expect other people to be able to identify it for you?
+
+The most basic way to define your problem is to explain what the expected behaviour is, and what the actual behaviour is. For example:
+
+> *I've written a function `sum` that is meant to return the sum of 2 numbers. When I do `sum(1, 1);` I'm expecting it to return 2, however it's returning 1.*
+
+Now whoever is reading my question knows what I'm trying to achieve and what is currently happening.
+
+I see a lot of questions where people describe their expected behaviour but omit their current behaviour, or the other way around.
+
+Also feel free to share screenshots if whatever you're working on has graphics, however **never** use screenshots for code, output, or errors that may display on the command line. Most people prefer that you copy and paste them instead, and it ensures it remains accessible for people with disabilities and translateable for people that speak different languages. Only use images when you need to if possible.
+
+## Share your problematic code
+
+If you can pinpoint the problematic code, then it's important that you share it. If it's too large introduce a link to a pastebin or github repository and explain what you're linking to so people know where to find your code, but remember not to link your entire repository as people don't have the time to be searching through your code to find the problem.
+
+In my previous example I would do the following:
+
+> *This is where `sum` is defined and how I'm calling `sum`:
+> 
+> ```javascript
+> function sum(a, b) {
+>     return a * b;
+> }
+> 
+> 
+> const result = sum(1, 1); // result == 1?
+> ```
+> 
+> *
+
+It also helps to introduce some comments if you don't have them already to add notes or explain some complex piece of code or maybe a function that is defined elsewhere.
```

Comment:
```suggestion
It also helps to introduce some comments if you don't have them already. These can add notes, explain a complex piece of code, or describe a function that is defined elsewhere.
```

Splitting up a longer sentence - I also try to avoid using either "and" or "or" multiple times in the same sentence when commas can be substituted instead, as that can be confusing to read.